frontend/src/app/new-claim/new-claim.component.ts

```diff
@@ -46,4 +46,36 @@ export class NewClaimComponent implements OnInit {
   sendNewClaim() {
     this.setClaim();
   }
+
+  hasEmptyFields(): boolean {
+    return this.areFieldsUntouched() || this.hasErroredFields();
+  }
+
+  areFieldsUntouched(): boolean {
+    return this.vin.untouched || this.color.untouched || this.model.untouched;
+  }
+
+  hasErroredFields(): boolean {
+    return (
+      this.vin.hasError('minlength') ||
+      this.vin.hasError('required') ||
+      this.color.hasError('required') ||
+      this.model.hasError('required')
+    );
+  }
+
+  checkIfSpecialChar(event) {
+    var eventCode;
+    eventCode = event.charCode;
+
+    // Only allow letters and numbers to be input.
+    return (
```

Comment:
Jesus dude, this makes my eyes bleed. :nauseated_face: 

Why is this better than, say this...

```html
<input type="text" pattern="[A-Za-z0-9]+" required />
```